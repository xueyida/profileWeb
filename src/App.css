@import '~antd/dist/antd.css';

.App {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
}

.App-header {
  height: 120px;
  background-color: #00bcd4;
  position: relative;
  display: flex;
  flex: 0 0 auto;
  /* align-content: center */
  justify-content: center;
}

.App-header-avatar {
  height: 120px;
  width: 120px;
  background-image: url('./images/image.jpg');
  background-size: cover;
  border-radius: 50%;
  position: absolute;
  top: 50%;
}


.App-header-top {
  width: 100%;
  display: flex;
  justify-content: space-between;
  padding: 5px 16px;
}

.App-header-iconWrap {
  display: flex;
  width: 80px;
  justify-content: space-around;
  font-size: 18px;
}

.App-content {
  background-color: #00a5bb;
  flex: 1 0 auto;
  color: #fff;
  padding-top: 60px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.App-content-subTitle {

}

.App-content-title {
  font-weight: 700;
  font-size: 16px;
  margin-top: 16px;
}

.App-content-description{
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  flex: 1 auto;
}

.App-content-description  .video{
  width: 45%;
}



.App-footer {
  height: 60px;
  background-color: #008fa1;
  text-align: center;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 16px;
  flex: 0 0 auto;
}

.App-footer-item {
  margin: 0 12px;
}

.App-footer-item a{
  color: #fff;
}



.App-footer-item-title{
  margin-left: 5px
}




@media screen and (max-width: 750px) {
    
  .App-content-description  .video{
    width: 100%;
  }

  .App-footer {
    flex-direction: column;
    height: 100px;
    font-size: 14px;
  }
  .App-footer-item {
    width: 250px;
    text-align: left;
  }

}