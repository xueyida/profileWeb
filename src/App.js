import React from 'react';
import { Result } from 'antd';

class App extends React.Component{
  render() {
    
    return(
      <div>
        <Result
          status="404"
          title="404"
          subTitle="Sorry, the page you visited does not exist."
        />
      </div>
    )
  }
}

export default App;
