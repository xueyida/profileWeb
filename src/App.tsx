import React from 'react';
import { Icon, Divider } from 'antd';
import './App.css';
import kobe from './images/rose.mp4';
import cover from './images/cover.png';

interface ListData {
  icon: string,
  title: string,
  isLink?: boolean
}

const data:Array<ListData> = [
  {
    icon: 'phone',
    title: '18234082909'
  },
  {
    icon: 'mail',
    title: '941820581@qq.com'
  },
  {
    icon: 'github',
    title: 'https://github.com/xueyida',
    isLink: true,
  }
]

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <div className="App-header-avatar"/>
        <div className="App-header-top">
          <div className="App-header-iconWrap">
            <Icon type="github" theme="outlined" />
            <Icon type="google" theme="outlined" />
            <Icon type="zhihu" theme="outlined" />
            {/* <Icon type="wechat" theme="outlined" /> */}
          </div>
        </div>
      </header>
      <div className="App-content">
        <div className="App-content-title">薛益达<Icon type="man" />男</div>
        <div className="App-content-subTitle">
          前端工程师
          <Divider type="vertical" />
          篮球爱好者
          <Divider type="vertical" />
          伪文艺青年
        </div>
        <div className="App-content-description">
          <video src={kobe}  className="video"  poster={cover} controls/>
        </div>
      </div>
      <footer className="App-footer">
        {
           data.map((item) => {
            return (
              <span className="App-footer-item">
                {
                  item.isLink?(
                    <a href={item.title} target="_blank">
                      <Icon type={item.icon} />
                      <span className="App-footer-item-title">{item.title}</span>
                    </a>
                  ):(
                    <>
                      <Icon type={item.icon} />
                      <span className="App-footer-item-title">{item.title}</span>
                    </>
                  )
                }
                
              </span>
            )
          })
        }
      </footer>
    </div>
  );
}

export default App;
