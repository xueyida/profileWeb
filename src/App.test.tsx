import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';
configure({ adapter: new Adapter() })


describe('snapshot', () => {
  it('snapshot', () => {
    const wrapRender = render(<App />);
    expect(
      wrapRender
    ).toMatchSnapshot();
  })
})

