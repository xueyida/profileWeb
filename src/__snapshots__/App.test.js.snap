// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "111",
        "next": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {},
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "123",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "123",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
      Object {
        "attribs": Object {},
        "children": Array [],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "123",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "123",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Object {
          "data": "111",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {},
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "123",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "123",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "data": "111",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "123",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "123",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {},
            "children": Array [],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "data": "111",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "123",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "123",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {},
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "data": "111",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
